src/OrchardCore.Modules/OrchardCore.Contents/Security/ContentTypeAuthorizationHandler.cs

```diff
@@ -33,89 +32,10 @@ protected override async Task HandleRequirementAsync(AuthorizationHandlerContext
 
             var contentItem = context.Resource as ContentItem;
 
-            Permission permission = null;
-
-            if (contentItem != null)
-            {
-                if (OwnerVariationExists(requirement.Permission) && HasOwnership(context.User, contentItem))
-                {
-                    permission = GetOwnerVariation(requirement.Permission);
-                }
-            }
-
-            var contentTypePermission = ContentTypePermissionsHelper.ConvertToDynamicPermission(permission ?? requirement.Permission);
-
-            if (contentTypePermission != null)
-            {
-                // The resource can be a content type name
-                var contentType = contentItem != null
-                    ? contentItem.ContentType
-                    : Convert.ToString(context.Resource.ToString())
-                    ;
-
-                if (!String.IsNullOrEmpty(contentType))
-                {
-                    permission = ContentTypePermissionsHelper.CreateDynamicPermission(contentTypePermission, contentType);
-                }
-            }
-
-            if (permission == null)
-            {
-                return;
-            }
-
-            // Lazy load to prevent circular dependencies
-            var authorizationService = _serviceProvider.GetService<IAuthorizationService>();
-
-            if (await authorizationService.AuthorizeAsync(context.User, permission))
+            if (await ContentTypeAuthorizationHelper.AuthorizeDynamicPermissionAsync(_httpContextAccessor.HttpContext, requirement.Permission, contentItem))
```

Comment:
This suggests that there was or is some issue with how the `ContentTypeAuthorizationHandler` was handling dynamic permissions.

I don't think we should have to be calling a helper in other places to check authorization, we should only ever need to call `AuthorizeAsync`. Otherwise we are fighting against the ASP.NET Authorization system, and lose extensibility.

But I don't understand the repro for the issue, so maybe some more info / repro

Reply:
Let me try something and I'll get back to you on that. I believe that the issue is that the ContentTypeAuthorizationHandler never gets called. Or if it does, the ContentItem passed = null and therefore doesn't handle the current ContentType we should be authorizing.

Reply:
Repro steps of the issue is : 

- Create a Content Type named "Acme".
- Create a new user Role named "SomeRole".
- Set "Edit own content" permission on that "Acme" content type in that "SomeRole" role.
- Create a new user named "NewUser" and set it's role to "SomeRole".
- Access the admin UI with that "NewUser" user and try to create a "Acme" content.

You will see that the button to save a draft is not displayed at all even though you have the permission to.
You can also try adding "Publish or unpublish Acme" permission on that "NewRole" and it won't show the publish button.

Reply:
Ok I confirm that the ContentTypeAuthorizationHandler gets called. So I'm making a change now to this PR to revert it back on calling the AuthorizationService.